packages/core/src/components/ui/Dropdown/Dropdown.List.js

```diff
@@ -17,26 +18,38 @@ const DropdownList = (props) => {
     options,
     optionRenderer,
     searchable,
-    searchModifier,
     onChange,
+    onSearchTermChange,
   } = props
   const [searchTerm, setSearchTerm] = useState('')
-  const filtered = _pickBy(options, (optionValue, optionKey) => {
-    const val = _isFunction(searchModifier) ? searchModifier(optionValue) : optionValue
-    const key = _isFunction(searchModifier) ? searchModifier(optionKey) : optionKey
-    const search = _isFunction(searchModifier) ? searchModifier(searchTerm) : searchTerm
 
-    return (
-      !searchTerm
-        || _includes(_toLower(key), _toLower(search))
-        || _includes(_toLower(val), _toLower(search))
-    )
-  })
-  const keys = _keys(filtered)
+  const filtered = useMemo(() => {
+    if (!searchable || _isEmpty(searchTerm) || _isFunction(onSearchTermChange)) {
+      return options
+    }
+
+    return _pickBy(options, (optionValue, optionKey) => !searchTerm
+          || _includes(_toLower(optionKey), _toLower(searchTerm))
+          || _includes(_toLower(optionValue), _toLower(searchTerm)))
+  },
+  [options, searchTerm, searchable, onSearchTermChange])
+
+  const keys = useMemo(() => _keys(filtered), [filtered])
 
   const handleSearchTermClick = (e) => {
     e.stopPropagation()
-    setSearchTerm(e.target.value)
+    const { value: _v } = e.target
+    if (_isFunction(onSearchTermChange)) {
+      onSearchTermChange(_v)
+    }
+    setSearchTerm(_v)
```

Comment:
use updateSearchTerm

```suggestion
    updateSearchTerm(_v)
```